lib/nostrum/api/base.ex

```diff
@@ -5,15 +5,27 @@ defmodule Nostrum.Api.Base do
 
   import Nostrum.Constants, only: [base_route: 0]
 
+  def request(conn, method, route, body, raw_headers, options) when is_list(options) do
+    request(conn, method, route, body, raw_headers, Map.new(options))
+  end
+
   def request(conn, method, route, body, raw_headers, options) do
-    full_route = "#{base_route()}#{route}"
     headers = process_request_headers(raw_headers)
     # Convert method from atom to string for `:gun`
     method =
       method
       |> Atom.to_string()
       |> String.upcase()
 
+    params = Map.get(options, :params, [])
+
+    query_string =
+      if params == [],
+        do: "",
+        else: "?" <> (Enum.map(params, fn {k, v} -> "#{k}=#{v}" end) |> Enum.join("&"))
```

Comment:
I think we can use [`URI.encode_query/1`](https://hexdocs.pm/elixir/1.12/URI.html#encode_query/2) here instead, right? That way this would also work with arguments with spaces in it.

Reply:
Ah, I knew there had to be a function for this.